components/brave_search/browser/brave_search_fallback.cc

```diff
@@ -0,0 +1,32 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_search/browser/brave_search_fallback.h"
+
+#include <utility>
+
+namespace brave_search {
+
+BraveSearchFallback::BraveSearchFallback() : weak_factory_(this) {}
```

Comment:
nit: I think we should use consistent naming with other similar mojo functionality and call this `BraveSearchHost`. Also in the future we might add other methods to this so no need to make it specific to `fetchBackupResults`.